sources/logdog/src/log_request.rs

```diff
@@ -196,21 +198,72 @@ where
             request: request.to_string()
         }
     );
-    let dest = tempdir.as_ref().join(request.filename);
-    let _ = fs::copy(&request.instructions, &dest).with_context(|| error::FileCopy {
-        request: request.to_string(),
-        from: request.instructions,
-        to: &dest,
+    let mut glob_paths = glob(request.instructions).context(error::ParseGlobPattern {
+        pattern: request.instructions.to_string(),
+    })?;
+    let multiple_files = glob_paths.count() > 1;
+    // iterator gets consumed in count, therefore re-create
+    glob_paths = glob(request.instructions).context(error::ParseGlobPattern {
+        pattern: request.instructions.to_string(),
     })?;
+    for entry in glob_paths {
+        if let Ok(path) = entry {
+            ensure!(path.is_file(), error::Directory { path });
```

Comment:
This feels like a bit of a harsh restriction because globs can wind up with directories fairly easily, and it can be hard to avoid them; you might need to use multiple overly specific globs just to avoid a directory when you actually want its contents anyway.

I think it'd be pretty easy to remove this restriction by using `walkdir` if the current glob result is a directory.  Something like:
```rust
for entry in WalkDir::new(&path) {
   let _ = fs::copy ...etc...
}
```

Reply:
Yeah I thought of this, there were two reason i thought of throwing error for directory pattern:
1. We do not need directory based pattern as of now. 
2. I did not want to introduce additional dependency unless it was needed.

I am happy to handle this requirement if you fee we should.

Reply:
> We do not need directory based pattern as of now.

We expect it, though, and allowing directories (instead of globs) was another primary option to address this problem, so I think it makes sense.

> I did not want to introduce additional dependency WalkDir unless it was needed.

We use walkdir in a number of places, so we're already building and trusting it, and it's quite nice to use! :)

Reply:
Okay, I will get started with this.